astropy/io/votable/tree.py

```diff
@@ -3978,6 +3978,8 @@ def to_xml(self, w, **kwargs):
             ]
             if kwargs["version_1_2_or_later"]:
                 element_sets.append(self.groups)
+                if not kwargs["version_1_4_or_later"]:
+                    del element_sets[0]
```

Comment:
I would find a comment here about the deprecation and undeprecation helpful. You do that below already, but there are already a lot of repetition in these methods, so the comment can be duplicated, too.

Reply:
Added in [f086bed](https://github.com/astropy/astropy/pull/17356/commits/f086bed3378177b11b1bb302596e7949daaf67d7)